clients/src/test/java/org/apache/kafka/common/network/SslTransportLayerTest.java

```diff
@@ -1014,12 +1014,6 @@ private void verifyInvalidReconfigure(ListenerReconfigurable reconfigurable,
         } catch (KafkaException e) {
             // expected exception
         }
-        try {
-            reconfigurable.reconfigure(invalidConfigs);
```

Comment:
We used to guarantee that reconfigure fails if configs are invalid. Is that no longer the case?

Reply:
It seems wasteful to do validation twice.  `validateReconfiguration` should be called before `reconfigure`.  Then `reconfigure` doesn't have to repeat all of the checks that were done in `validateReconfiguration`.

Reply:
The file may be modified asynchronously and the keystore and truststore loaded from file during `reconfigure` may not be the same as the one used during `validateReconfiguration`. Unless we guarantee that we save and use the stores loaded during validation, we have to ensure that `reconfigure` doesn't break the broker.

Reply:
Even if `reconfigure` re-does the validation, the file could still be changed in between the validation and when we start using it.  There is no way to fully close the race condition except by copying the file to a separate location that only we control.  That level of paranoia seems out of scope here.

It's also not clear that this level of paranoia is needed at all if people adopt a sane policy like "don't modify the truststore / keystore files in palce"  If those files are modified in place, there will always be some short time period the modification and Kafka re-validating that the configuration still works.  And there will also be no working configuration to fall back on in that case.

Reply:
@cmccabe `reconfigure` was testing that the keystores and truststores loaded from files by `reconfigure` and used to create the new SSL context are actually valid. If the file changes after that, it is fine since the new files will not be used for creating a new context until the next admin client request to update. For this particular invocation of `reconfigure`, we need to guarantee that SSLContext is updated only if the keystore/truststore used to the create the new SSLContext are valid.

Even though we tried to push back on in-place updates of keystore/truststore, it seems to be what everyone wants to do. And hence supporting this without any timing issues is critical.

Reply:
@cmccabe Can we keep this test now since we validate on reconfiguration?

Reply:
I changed it back.